### PR TITLE
Add support for react hot loading with HOT=1 environment variable

### PR DESCRIPTION
**Pull request by pcardune**
This runs webpack-dev-server on port 3001 and makes it proxy any requests it doesn't know how to serve back to rails.

with the current code, you just do:

```
HOT=1 npm start
```

and then use 3001 instead of 3000. Any changes to react components you make will get compiled and "hot-loaded" onto the page without needing a page refresh. This is great when you are making UI tweaks.

### CODE REVIEW

File: apps/Gruntfile.js [hunk 1]
```diff
@@ -685,10 +703,15 @@ module.exports = function (grunt) {
   grunt.registerTask('dev', [
     'prebuild',
     'newer:sass',
-    'concurrent:watch',
+    'webpack-dev-server:watch',
+//    'concurrent:watch',
```
Comment (pcardune) [inline apps/Gruntfile.js]: probably want to go back to concurrent:watch here.

File: apps/Gruntfile.js [hunk 2]
```diff
@@ -685,10 +703,15 @@ module.exports = function (grunt) {
...
     'postbuild',
   ]);
 
+  grunt.registerTask('dev-server', [
+    'webpack-dev-server:watch',
+  ]);
```
Comment (pcardune) [inline apps/Gruntfile.js]: probably want to ditch this task in favor of just using dev. Or would it be better for people to choose between running webpack-dev-server vs regular webpack?

File: apps/webpack.js [hunk 3]
```diff
@@ -35,6 +35,11 @@ var baseConfig = {
           path.resolve(__dirname, 'static'),
         ],
         loader: "url-loader?limit=1024",
+      },
+      {
+        test: /\.jsx?$/,
+        loader: 'react-hot',
+        include: [path.resolve(__dirname, 'src')]
```
Comment (pcardune) [inline apps/webpack.js]: need to make the use of this loader optional, so it doesn't get used for production builds.

**Comment by pcardune** [comment]
Ok, this is ready to actually be reviewed

**Comment by Bjvanminnen** [comment]
Cool stuff. lgtm

**Comment by bcjordan** [comment]
Very cool—@pcardune anecdotally, how fast is the iteration loop from code change to UI update now (in the happy-path where the reload goes well)?

**Comment by pcardune** [comment]
not as fast as I'd like it to be. It depends on what code you are changing and how many of the assets need to get rebuilt as a result. The way our code is split between different assets does not make this super optimal at the moment :(

**Pull request merged**